src/library_html5.js

```diff
@@ -935,17 +935,41 @@ var LibraryHTML5 = {
 
   $fillOrientationChangeEventData__deps: ['$screenOrientation'],
   $fillOrientationChangeEventData: (eventStruct) => {
-    var orientations  = ["portrait-primary", "portrait-secondary", "landscape-primary", "landscape-secondary"];
-    var orientations2 = ["portrait",         "portrait",           "landscape",         "landscape"];
-
-    var orientationString = screenOrientation();
-    var orientation = orientations.indexOf(orientationString);
-    if (orientation == -1) {
-      orientation = orientations2.indexOf(orientationString);
+    // OrientationType enum
+    var orientationsType1 = ['portrait-primary', 'portrait-secondary', 'landscape-primary', 'landscape-secondary'];
+    // alternative selection from OrientationLockType enum
+    var orientationsType2 = ['portrait',         'portrait',           'landscape',         'landscape'];
+
+    var orientationIndex = -1;
+    var orientationAngle =  0;
+    var screenOrientObj  = screenOrientation();
+    if (screenOrientObj) {
+      var orientationType;
+      if (typeof screenOrientObj === 'object') {
+        orientationType  = screenOrientObj.type;
+        orientationAngle = screenOrientObj.angle;
+      } else {
+        // Edge only supports the older string type until 2020 (pre-Chrome)
+        orientationType  = String(screenOrientObj);
+      }
+      orientationIndex = orientationsType1.indexOf(orientationType);
+      if (orientationIndex == -1) {
+        orientationIndex = orientationsType2.indexOf(orientationType);
+      }
+      if (orientationIndex != -1) {
+        orientationIndex = 1 << orientationIndex;
```

Comment:
Maybe we could add a

```c
#define EMSCRIPTEN_ORIENTATION_INVALID  0
```
into html5.h

and then have this callback send that value in the orientation index if not detected, rather than sending `-1`. That way callers will be able to react in their code accordingly.

Reply:
I agree it would be correct, but wouldn't that break existing code?

Reply:
I had a think, and since the code hasn't worked for a long time (always returning `-1`) then a breaking change doesn't make a difference.

I added `EMSCRIPTEN_ORIENTATION_UNSUPPORTED`, made sure this is always returned instead of `-1`, and added it to the docs.